sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/decimalExpressions.scala

```diff
@@ -45,9 +45,15 @@ case class UnscaledValue(child: Expression) extends UnaryExpression {
  * Note: this expression is internal and created only by the optimizer,
  * we don't need to do type check for it.
  */
-case class MakeDecimal(child: Expression, precision: Int, scale: Int) extends UnaryExpression {
+case class MakeDecimal(
+    child: Expression,
+    precision: Int,
+    scale: Int,
+    nullOnOverflow: Boolean) extends UnaryExpression {
 
-  private val nullOnOverflow = !SQLConf.get.ansiEnabled
```

Comment:
Is this change different to previous one?

Original `nullOnOverflow` is a `val` that is initialized during constructing of `MakeDecimal` and won't be changed then, no?

Reply:
In my understanding what we want to avoid is that `nullOnOverflow` changes during expression transformation: https://github.com/apache/spark/pull/27683#issuecomment-590317541

Reply:
Oh, I see. Makes sense now.

Reply:
Can we add a test for this?

Reply:
Yea, better to add tests if we can.

Reply:
All these 8 in https://issues.apache.org/jira/browse/SPARK-30893 should be tested? I can try to add some kind of test in a follow-up PR next week.

Reply:
Yea, I think we should do where possible.

Reply:
We should have tests to for such behaviors. Otherwise, we could possibly change it unintentionally.